### PR TITLE
nxos_snmp_community: Use running config for module operations instead of cli commands. 

### PR DESCRIPTION
**Pull request by GGabriele**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
nxos_snmp_community

##### ANSIBLE VERSION
```
ansible 2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
Changed how module work to make it use running config for module operations instead of cli commands. Referring to #5214

### CODE REVIEW

**Comment by GGabriele** [comment]
@privateip  The error is due to `version_added: 2.2` on docstring while it expects 2.3, but actually those params have been introduced in 2.2.

**Comment by gregdek** [comment]
Thanks @GGabriele for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_snmp_community.py [hunk 1]
```diff
@@ -54,6 +54,29 @@
...
         default: present
         choices: ['present','absent']
+    include_defaults:
+        description:
+            - Specify to use or not the complete runnning configuration
+              for module operations.
+        required: false
+        default: false
+        choices: ['true','false']
+        version_added: 2.2
```
Comment (gundalow) [inline network/nxos/nxos_snmp_community.py]: 2.3

File: network/nxos/nxos_snmp_community.py [hunk 2]
```diff
@@ -54,6 +54,29 @@
...
+    config:
+        description:
+            - Configuration string to be used for module operations. If not
+              specified, the module will use the current running configuration.
+        required: false
+        default: null
+        version_added: 2.2
```
Comment (gundalow) [inline network/nxos/nxos_snmp_community.py]: 2.3

File: network/nxos/nxos_snmp_community.py [hunk 3]
```diff
@@ -54,6 +54,29 @@
...
+    save:
+        description:
+            - Specify to save the running configuration after
+              module operations.
+        required: false
+        default: false
+        choices: ['true','false']
+        version_added: 2.2
```
Comment (gundalow) [inline network/nxos/nxos_snmp_community.py]: 2.3

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Comment by GGabriele** [comment]
ready_for_review

**Review by gundalow — APPROVED** [review]

**Comment by gregdek** [comment]
Thanks @GGabriele. To the current maintainers, @jedelman8, @GGabriele, @privateip, @gundalow, @qalthos please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
@privateip @Qalthos Please review and if you are happy merge

**Pull request closed without merging**